modin/pandas/groupby.py

```diff
@@ -631,21 +654,35 @@ def ngroup(self, ascending=True):
         return self._default_to_pandas(lambda df: df.ngroup(ascending))
 
     def nunique(self, dropna=True):
-        return self._apply_agg_function(lambda df: df.nunique(dropna))
+        result = self._apply_agg_function(lambda df: df.nunique(dropna))
```

Comment:
I just wonder if those two cases (of resetting the index and setting its name) should be done via decorators to reduce code duplication and ease maintainership

Reply:
I agree, that it reduces duplication and set one entry point for this checks. Starting from 16a3ff1c1e7a8e8ff327e2644070690acecef786 wrappers is used.